source/MaterialXCore/Document.cpp

```diff
@@ -97,6 +97,18 @@ class Document::Cache
                         portElementMap.emplace(portElem->getQualifiedName(nodeName), portElem);
                     }
                 }
+                else
+                {
+                    const string& nodegraphName = elem->getAttribute(PortElement::NODE_GRAPH_ATTRIBUTE);
```

Comment:
For readability, I would recommend moving this call to the top of the `for` loop, right after the storage of `nodeDefString`, so that the parallel between these two calls is clear.